pylint/extensions/unassigned_function_call.py

```diff
@@ -0,0 +1,91 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/pylint-dev/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/pylint-dev/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Looks for unassigned function/method calls that have non-nullable return type."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import astroid
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.typecheck import TypeChecker
+from pylint.checkers.utils import only_required_for_messages, safe_infer
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class UnusedReturnValueChecker(BaseChecker):
+    name = "unused_return_value"
+    msgs = {
+        "W5486": (
+            "Function returned value which is never used",
+            "unused-return-value",
```

Comment:
I just realized that we have a message for expression that aren't assigned : https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/expression-not-assigned.html. Maybe grouping those two in the same checker and calling this one the same way would make sense:
```suggestion
            "function-return-not-assigned",
```

Reply:
Relevant issues:
- https://github.com/pylint-dev/pylint/issues/8129
- https://github.com/pylint-dev/pylint/issues/2380 
 
We might want to check some ternary construct.

Reply:
Thanks for the review. Docs and examples added, switched to functional tests, message was renamed to `function-return-not-assigned`.